flytekit/annotated/promise.py

```diff
@@ -1,36 +1,65 @@
 import collections
 from typing import Union, Dict, Any, List, Tuple
 
+from flytekit.annotated.context_manager import FlyteContext
 from flytekit import engine as flytekit_engine
 from flytekit.common.promise import NodeOutput as _NodeOutput
-from flytekit.models import literals as _literal_models, interface as _interface_models
+from flytekit.models import literals as _literal_models, interface as _interface_models, types as _type_models
 
 
-def translate_inputs_to_literals(ctx, input_kwargs: Dict[str, Any], interface: _interface_models.TypedInterface):
+def translate_inputs_to_literals(ctx: FlyteContext, input_kwargs: Dict[str, Any],
+                                 interface: _interface_models.TypedInterface) -> Dict[str, _literal_models.Literal]:
     """
     When calling a task inside a workflow, a user might do something like this.
 
-    def my_wf(in1: int) -> int:
-        a = task_1(in1=in1)
-        b = task_2(in1=5, in2=a)
-        return b
+        def my_wf(in1: int) -> int:
+            a = task_1(in1=in1)
+            b = task_2(in1=5, in2=a)
+            return b
 
     If this is the case, when task_2 is called in local workflow execution, we'll need to translate the Python native
     literal 5 to a Flyte literal.
+
+    More interesting is this:
+
+        def my_wf(in1: int, in2: int) -> int:
+            a = task_1(in1=in1)
+            b = task_2(in1=5, in2=[a, in2])
+            return b
+
+    Here, in task_2, during execution we'd have a list of Promises. We have to make sure to give task2 a Flyte
+    LiteralCollection (Flyte's name for list), not a Python list of Flyte literals.
     """
-    # TODO: Are there any pass-by-reference considerations we need to think about?  I don't think so. Write unit
-    #  tests to be sure.
+
+    def extract_value(ctx: FlyteContext, input_val: Any, flyte_literal_type: _type_models.LiteralType) -> Any:
```

Comment:
why is the return type Any? It seems that the return is always a Literal?

Reply:
cuz i was wrong.